### ISSUE TITLE
[bug] rock fails to install using git repo

### ISSUE THREAD

**Issue by Tieske** [luarocks/luarocks#1154]
Referring to a git repo will pack and upload, but fails to `luarocks install`. Locally a `luarocks make` will succeed on the exact same rockspec. Not using a git repo, but using a tarball works as a workaround.

To reproduce 2 rockspecs: [this works](https://github.com/Tieske/timerwheel.lua/blob/6baf5db703c1abd776c2f4b1a337a8d04226d43a/rockspecs/timerwheel-0.1.0-2.rockspec#L4-L5), but [this doesn't](https://github.com/Tieske/timerwheel.lua/blob/6baf5db703c1abd776c2f4b1a337a8d04226d43a/rockspecs/timerwheel-0.1.0-1.rockspec#L4-L5)

Weird part is that packing and uploading the broken rockspec actually works, just as `make`, it is only `install` that fails, both on the rockspec itself, as well as on the packed rock.

My hunch is that it is due to some path handling, the repo is called `timerwheel.lua` and that might confuse some directory handling routines to think it is actually a file maybe?

**Comment by osch** [comment]
What is your current directory when you do `make`? My impression is, that when using `make` the paths in the rockspec are evaluated by Luarocks relative to the current directory. When using `install` the paths are evaluated relative to the location given in the `source.dir` attribute of the rockspec.

**Comment by Tieske** [comment]
since the difference between working/not-working is only using a git or tar url, I don't think that make vs install is the issue.

**Comment by hishamhm** [comment]
> My hunch is that it is due to some path handling, the repo is called timerwheel.lua and that might confuse some directory handling routines to think it is actually a file maybe?

That is very much a possibility. Can you try 

```
  url = "git://github.com/Tieske/timerwheel.lua/",
```

(with a trailing slash) ?

**Comment by hishamhm** [comment]
Ok, so I did try it and:

1. yes, it is because of the `.lua` extension
2. yes, adding a slash works
3. adding `source.dir = "timerwheel.lua"` also works
4. most interestingly, adding `rockspec_format = "3.0"` also works! Fixing this bug is technically a breaking change, so it was already fixed in the switch to LuaRocks 3.0. So, you get the desired behavior when switching to the 3.0 format, but the old format still gets the old behavior for strict compatibility with custom fetch protocols.

So, tl;dr, add `rockspec_format = "3.0"` to the rockspec and problem solved. :)

**Comment by Tieske** [comment]
yep, just tried as well, the extra slash works.

Ok, so forward fix. Thx
